File: common/src/main/java/com/mall/common/entity/Product.java
package com.mall.common.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体 — product-service 管理
 */
@Data
@TableName("mall_product")
public class Product {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;                // 商品名称
    private String description;         // 描述
    private BigDecimal price;           // 价格
    private Integer stock;              // 库存
    private String imageUrl;            // 图片URL
    private String category;            // 分类
    private Integer status;             // 1上架 0下架
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/product.ts
import request from './request'

export interface ProductItem {
  id: number; name: string; description: string
  price: number; stock: number; imageUrl: string; category: string
  status: number; createTime: string
}
export interface ProductPage { records: ProductItem[]; total: number; size: number; current: number }

export function listProducts(page = 1, size = 10, category?: string) {
  return request.get('/product/list', { params: { page, size, category } })
}
export function getProductDetail(id: number) {
  return request.get(`/product/detail/${id}`)
}
